### PR TITLE
Add TLS support for vs/vsr upstreams

### LINKED ISSUE #620: VirtualServer with SSL services?
**Is your feature request related to a problem? Please describe.**
Currently, I don't see there is an option to create a `VirtualServer` with secured service. When define the upstream with port 443, I see this error:
image

**Describe the solution you'd like**
Be able to use SSL services in `VirtualServer`, maybe use the same annotation as in `Ingress`?
```yml
apiVersion: k8s.nginx.org/v1alpha1
kind: VirtualServer
metadata:
  name: secure-app-virtual-server
  annotations:
    nginx.org/ssl-services: "secure-app"
spec:
  host: example.com
  tls:
    secret: secure-app-certs
  upstreams:
  - name: secure-app
    service: secure-app
    port: 443
  routes:
  - path: /
    upstream: secure-app
```

### PR DESCRIPTION
**Pull request by Rulox**
### Proposed changes
Closes #620

Add the possibility to configure TLS for Upstreams in VS/VSR. This will redirect the requests to the upstream using `https` or `http` depending on the user's configuration.

### CODE REVIEW

File: docs/virtualserver-and-virtualserverroute.md [hunk 1]
```diff
@@ -197,6 +200,12 @@ send-timeout: 30s
 `connect-timeout` | The timeout for establishing a connection with an upstream server. See the [proxy_connect_timeout](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_connect_timeout) directive. The default is specified in the `proxy-connect-timeout` ConfigMap key. | `string` | No
 `read-timeout` | The timeout for reading a response from an upstream server. See the [proxy_read_timeout](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_read_timeout) directive.  The default is specified in the `proxy-read-timeout` ConfigMap key. | `string` | No
 `send-timeout` | The timeout for transmitting a request to an upstream server. See the [proxy_send_timeout](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout) directive. The default is specified in the `proxy-send-timeout` ConfigMap key. | `string` | No
+| `tls` | The TLS configuration for the Upstream. | `tls` | No |
```
Comment (pleshakov) [inline docs/virtualserver-and-virtualserverroute.md]: `tls` in the type column should be a link . Please take a look at the splits and rules rows of the VirtualServer.Route

File: docs/virtualserver-and-virtualserverroute.md [hunk 2]
```diff
@@ -197,6 +200,12 @@ send-timeout: 30s
...
+
+### Upstream.TLS
+| Field | Description | Type | Required |
+| ----- | ----------- | ---- | -------- |
+| `enable` | Enables HTTPS for requests to upstream servers. The default is `False`, meaning that HTTP will be used. | `Boolean` | No |
```
Comment (pleshakov) [inline docs/virtualserver-and-virtualserverroute.md]: Should we change `Boolean` to `boolean` for consistency? All other types start with a lowercase letter.

File: internal/configs/virtualserver.go [hunk 3]
```diff
@@ -258,6 +258,11 @@ func upstreamHasKeepalive(upstream conf_v1alpha1.Upstream, cfgParams *ConfigPara
 }
 
 func generateLocation(path string, upstreamName string, upstream conf_v1alpha1.Upstream, cfgParams *ConfigParams) version2.Location {
+	proxyPassProtocol := "http"
```
Comment (pleshakov) [inline internal/configs/virtualserver.go]: I suggest having a function like `generateProxyPassProtocol` that accepts an upstream as an argument. I suggest writing a unit test for that function.

File: docs/virtualserver-and-virtualserverroute.md (hunk 2 above)
Comment (Rulox) [reply docs/virtualserver-and-virtualserverroute.md]: Yeah makes sense

File: internal/configs/virtualserver.go (hunk 3 above)
Comment (Rulox) [reply internal/configs/virtualserver.go]: I forgot to add tests yes. I don't see a reason to split 3 lines of code into a different function for now.

**Pull request merged**